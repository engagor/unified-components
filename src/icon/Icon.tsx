import React, { ReactElement } from 'react';
import { ClarabridgeIconsId } from '@clarabridge/unified-icons/src/types/clarabridge-icons';

export type IconProps = {
    iconName: ClarabridgeIconsId;
    className?: string;
};

export default function Icon(props: IconProps): ReactElement {
    return <span className={`en-icon en-icon-${props.iconName} ${props.className ?? ''}`} aria-hidden="true" />
}
