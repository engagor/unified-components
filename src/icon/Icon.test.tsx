import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

it('renders correctly', async () => {
    const element = document.createElement('div');

    const icon = await renderer
        .create(<Icon iconName="add" />)
        .toJSON();

    await expect(icon).toMatchSnapshot();
    expect(element).toMatchSnapshot();
});
