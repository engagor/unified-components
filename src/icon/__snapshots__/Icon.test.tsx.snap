// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  aria-hidden="true"
  className="en-icon en-icon-add "
/>
`;

exports[`renders correctly 2`] = `<div />`;
